picard/formats/__init__.py

```diff
@@ -39,17 +39,44 @@ def supported_formats():
     return formats
 
 
+def guess_format(filename, options=_formats):
+    """Select the best matching file type amongst supported formats."""
+    results = []
+    with file(filename, "rb") as fileobj:
+        header = fileobj.read(128)
+        # Calls the score method of a particular format's associated filetype
+        # and assigns a positive score depending on how closely the fileobj's header matches
+        # the header for a particular file format.
+        results = [(option._File.score(filename, fileobj, header), option.__name__, option)
+                   for option in options
+                   if hasattr(option, "_File")
+                   and option._File is not None]
+    if results:
+        results.sort()
+        if results[-1][0] > 0:
+            # return the format with the highest matching score
+            return results[-1][2](filename)
+
+    # No positive score i.e. the fileobj's header did not match any supported format
+    return None
+
+
 def open(filename):
     """Open the specified file and return a File instance with the appropriate format handler, or None."""
     i = filename.rfind(".")
```

Comment:
Those lines are detecting extension, if none is found at all it will return `None`, but imho we should still try magic number in this case.
So move those lines after `if not audio_file`